source/s3_auto_ranged_put.c

```diff
@@ -1102,61 +1259,71 @@ static void s_s3_auto_ranged_put_request_finished(
             AWS_FATAL_ASSERT(part_number > 0);
             size_t part_index = part_number - 1;
             struct aws_string *etag = NULL;
+            bool request_is_noop = request->is_noop != 0;
 
-            if (error_code == AWS_ERROR_SUCCESS) {
-                /* Find the ETag header if it exists and cache it. */
-                struct aws_byte_cursor etag_within_quotes;
+            if (!request_is_noop) {
+                if (error_code == AWS_ERROR_SUCCESS) {
+                    /* Find the ETag header if it exists and cache it. */
+                    struct aws_byte_cursor etag_within_quotes;
 
-                AWS_ASSERT(request->send_data.response_headers);
+                    AWS_ASSERT(request->send_data.response_headers);
 
-                if (aws_http_headers_get(
-                        request->send_data.response_headers, g_etag_header_name, &etag_within_quotes) !=
-                    AWS_OP_SUCCESS) {
-                    AWS_LOGF_ERROR(
-                        AWS_LS_S3_META_REQUEST,
-                        "id=%p Could not find ETag header for request %p",
-                        (void *)meta_request,
-                        (void *)request);
+                    if (aws_http_headers_get(
+                            request->send_data.response_headers, g_etag_header_name, &etag_within_quotes) !=
+                        AWS_OP_SUCCESS) {
+                        AWS_LOGF_ERROR(
+                            AWS_LS_S3_META_REQUEST,
+                            "id=%p Could not find ETag header for request %p",
+                            (void *)meta_request,
+                            (void *)request);
 
-                    error_code = AWS_ERROR_S3_MISSING_ETAG;
-                } else {
-                    /* The ETag value arrives in quotes, but we don't want it in quotes when we send it back up
-                     * later, so just get rid of the quotes now. */
-                    etag = aws_strip_quotes(meta_request->allocator, etag_within_quotes);
+                        error_code = AWS_ERROR_S3_MISSING_ETAG;
+                    } else {
+                        /* The ETag value arrives in quotes, but we don't want it in quotes when we send it back up
+                         * later, so just get rid of the quotes now. */
+                        etag = aws_strip_quotes(meta_request->allocator, etag_within_quotes);
+                    }
+                }
+                if (error_code == AWS_ERROR_SUCCESS && meta_request->progress_callback != NULL) {
```

Comment:
Oh, we need to update the documentation about progress callback. https://github.com/awslabs/aws-c-s3/blob/main/include/aws/s3/s3_client.h#L422-L428

It says it only supports for copyObject, that's not true.